@here/harp-datasource-protocol/lib/InterpolatedProperty.ts

```diff
@@ -103,3 +63,179 @@ export function isInterpolatedProperty<T>(p: any): p is InterpolatedProperty<T>
     }
     return false;
 }
+
+/**
+ * Get the value of the specified property at the given zoom level, represented as a `number` value.
+ *
+ * @param property Property of a technique.
+ * @param level Display level the property should be rendered at.
+ * @param pixelToMeters Optional pixels to meters conversion factor (needed for proper
+ * interpolation of `length` values).
+ *
+ */
+export function getPropertyValue<T>(
+    property: InterpolatedProperty<T> | MaybeInterpolatedProperty<T>,
+    level: number,
+    pixelToMeters: number = 1.0
+): number {
+    if (isInterpolatedPropertyDefinition<T>(property)) {
+        throw new Error("Cannot interpolate a InterpolatedPropertyDefinition.");
+    } else if (!isInterpolatedProperty(property)) {
+        if (typeof property !== "string") {
+            return (property as unknown) as number;
+        } else {
+            if (
```

Comment:
I believe something like this would avoid this code hell (duplicating code & long lines):

```js
const matchedNumeralType = [
  StringEncodedNumeralType.Meters,
  StringEncodedNumeralType.Pixels
]
  .find(numeralType => StringEncodedNumeralFormats[numeralFormat].regExp.test(property));

const matchedNumeralFormat = StringEncodedNumeralFormats[matchedNumeralType];

if (matchedNumeralFormat !== undefined) {
  return matchedNumeralFormat.decoder(property)[0] * (
    matchedNumeralType === StringEncodedNumeralType.Pixels ? pixelToMeters : 1
  );
} else {
 // ... 
}
```

I do believe this could help.

Reply:
(Part 2)
And a possible switch-case condition as well:

```js

switch (matchedNumeralType) {
  case StringEncodedNumeralType.Meters:
    return matchedNumeralFormat.decoder(property)[0];
  case StringEncodedNumeralType.Pixels:
    return matchedNumeralFormat.decoder(property)[0] * pixelToMeters;
  default:
    // ...
  
}
```

Reply:
Good suggestion, applied in last patchset!